src/.clang-tidy

```diff
@@ -5,24 +5,13 @@ bugprone-use-after-move,
 misc-unused-using-decls,
 modernize-use-default-member-init,
 modernize-use-nullptr,
-performance-for-range-copy,
-performance-move-const-arg,
-performance-unnecessary-copy-initialization,
-readability-redundant-declaration,
-readability-redundant-string-init,
-'
-WarningsAsErrors: '
-bugprone-argument-comment,
-bugprone-use-after-move,
-misc-unused-using-decls,
-modernize-use-default-member-init,
-modernize-use-nullptr,
-performance-for-range-copy,
-performance-move-const-arg,
-performance-unnecessary-copy-initialization,
+performance-*,
+-performance-no-int-to-ptr,
+-performance-unnecessary-value-param,
```

Comment:
Is there a reason why they these two are not enabled?

Reply:
> Is there a reason why they these two are not enabled?

The required fixes seem too invasive for this PR.